### ISSUE TITLE
Cannot index typed dictionary using (string) enum literal

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#4036]
**Describe the bug**
A `TypedDict` with keys which correspond to a `StrEnum`'s members cannot be indexed using the enum.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That Pyright would treat the `StrEnum`'s members' values as string literals.

**Screenshots or Code**
```python
# pyright: strict

from enum import StrEnum
from typing import Final, TypedDict

A = 'a'

class Foo(StrEnum):
    a = 'a'

class Bar(TypedDict):
    a: int

bar: Bar = {A: 1}  # Ok
bar2: Bar = {Foo.a: 1}  # Error: Expression of type "dict[Foo, int]" cannot be assigned to declared type "Bar"
bar3: Final[Bar] = {Foo.a: 1}  # As above
bar[Foo.a]  # Error: Could not access item in TypedDict;  "Literal[Foo.a]" is not a string literal

```

**VS Code extension or command-line**
1.1.274

**Additional context**
I'm not sure if this a bug report or a feature request; I was mostly expecting it to work [like it does in TS](https://www.typescriptlang.org/play?#code/KYOwrgtgBAYg9nKBvAUFdUCGUC8UDkm+ANCgL4ooCWIALsAE4BmmAxsFAEKYPJoaYAXFHAQARowDc5SqzggAzrShiew7rzxIoAbXhwAdJgC6wgIxQKQA) with interfaces but I realise that `TypedDict` is rather limited by comparison, while Python enums are vastly more intricate.

**Comment by erictraut** [comment]
I don't think this is a bug. A `TypedDict` requires keys of type `str`, not a subclass of `str`. 

This will work:

```python
bar2: Bar = {Foo.a.value: 1}
```

**Comment by erictraut** [comment]
Making `TypedDict` work with keys of other types like `StrEnum` would require a change to the `TypedDict` specification. If you want to pursue that, you could bring it up for discussion in the [python/typing](https://github.com/python/typing/discussions) or [typing-sig](https://mail.python.org/archives/list/typing-sig@python.org/) forums.

**Comment by erictraut** [comment]
When I run the latest version of mypy on your code above, it emits the error `Expected TypedDict key to be string literal` in the same three locations where pyright reports an error. That's what I would expect.

**Comment by layday** [comment]
I was (mostly) wrong. It looks like mypy only supports indexing with a string enum - the last line doesn't produce an error in mypy.  `bar: Bar = {A: 1}` does, however, because `A` is not marked as final.

**Comment by olejorgenb** [comment]
https://github.com/python/typing/discussions/1721  (autolinking doesn't work for discussions?)
